src/Application/Features/Teams/Commands/AddHeroToTeam/AddHeroToTeamCommand.cs

```diff
@@ -0,0 +1,52 @@
+﻿using SSW.CleanArchitecture.Application.Common.Exceptions;
+using SSW.CleanArchitecture.Application.Common.Interfaces;
+using SSW.CleanArchitecture.Domain.Heroes;
+using SSW.CleanArchitecture.Domain.Teams;
+
+namespace SSW.CleanArchitecture.Application.Features.Teams.Commands.AddHeroToTeam;
+
+public sealed record AddHeroToTeamCommand(Guid TeamId, Guid HeroId) : IRequest;
+
+// ReSharper disable once UnusedType.Global
+public sealed class AddHeroToTeamCommandHandler(IApplicationDbContext dbContext)
+    : IRequestHandler<AddHeroToTeamCommand>
+{
+    public async Task Handle(AddHeroToTeamCommand request, CancellationToken cancellationToken)
+    {
+        var teamId = new TeamId(request.TeamId);
+        var heroId = new HeroId(request.HeroId);
+
+        var team = dbContext.Teams
+            .WithSpecification(new TeamByIdSpec(teamId))
+            .FirstOrDefault();
+
+        if (team is null)
```

Comment:
I tend to put these checks (team and hero) be in the validator

Reply:
I go back and forward on this.  I used to be of the mind that absolutely all validation should go in the validator.  No I believe that it should only be input validation.  Checking of valid DB IDs, and business logic will happen elsewhere.